app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-link-click-tracker.js

```diff
@@ -0,0 +1,156 @@
+      this.internalLinksDomain = 'www.gov.uk/'
+      document.querySelector('body').addEventListener('click', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('contextmenu', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('mousedown', this.handleMousedown.bind(this))
+    },
+
+    handleClick: function (event) {
+      var element = event.target
+
+      if (element.tagName !== 'A') {
+        element = element.closest('a')
+      }
+
+      if (!element) {
+        return
+      }
+
+      var clickData = {}
+      var href = element.getAttribute('href')
+
+      if (this.isMailToLink(href)) {
+        clickData.type = 'email'
+        clickData.external = 'true'
+      } else if (this.isDownloadLink(href)) {
+        clickData.type = 'download'
+        clickData.external = 'false'
+      } else if (this.isExternalLink(href)) {
+        clickData.type = 'generic link'
+        clickData.external = 'true'
+      }
+
+      if (Object.keys(clickData).length > 0) {
+        clickData.event_name = 'navigation'
+        clickData.text = element.textContent.trim()
+        clickData.url = href
+        clickData.link_method = this.getClickType(event)
+
+        var schema = new window.GOVUK.analyticsGA4.Schemas().eventSchema()
+        schema.event = 'analytics'
+
+        // get attributes from the clickData object to send to GA
+        // only allow it if it already exists in the schema
+        for (var key in clickData) {
+          if (schema.event_data[key]) {
+            schema.event_data[key] = clickData[key]
+          }
+        }
+
+        this.trackClickEvent(schema)
+      }
+    },
+
+    getClickType: function (event) {
+      switch (event.type) {
+        case 'click':
+          if (event.ctrlKey) {
+            return 'ctrl click'
+          } else if (event.metaKey) {
+            return 'command/win click'
+          } else {
+            return 'primary click'
+          }
+        case 'mousedown':
+          return 'middle click'
+        case 'contextmenu':
+          return 'secondary click'
+      }
+    },
+
+    handleMousedown: function (event) {
+      // 1 = middle mouse button
+      if (event.button === 1) {
+        this.handleClick(event)
+      }
+    },
+
+    isMailToLink: function (href) {
+      if (!href) {
+        return false
+      }
+
+      if (href.length < 7) {
+        return false
+      }
+      return href.substring(0, 7) === 'mailto:'
+    },
+
+    isDownloadLink: function (href) {
+      var assetsDomain = 'assets.publishing.service.gov.uk/'
+      var uploadsPath = '/government/uploads/'
+      if (!href) {
+        return false
+      }
+
+      if (this.hrefPointsToDomain(href, assetsDomain)) {
+        return true
+      }
+
+      var isInternalLink = this.hrefPointsToDomain(href, this.internalLinksDomain) || this.hrefPointsToDomain(href, this.internalLinksDomain.replace('www.', ''))
```

Comment:
I'd suggest having a second variable for `this.internalLinksDomain.replace('www.', '')` rather than recalculating it each click e.g. `this.internalLinksDomainWithoutWww = 'gov.uk/'`